### PR TITLE
NEW added application metadata

### PR DESCRIPTION
**Pull request by rodgomes**

### CODE REVIEW

File: katka/admin.py [hunk 1]
```diff
@@ -58,6 +58,6 @@ class SCMStepRunAdmin(WithUsernameAdminModel):
     fields = ('slug', 'name', 'stage', 'status', 'output', 'scm_pipeline_run')
 
 
-@admin.register(SCMRelease)
```
Comment (JeroennC) [inline katka/admin.py]: Why did you remove this from admin?

File: katka/models.py [hunk 2]
```diff
@@ -122,3 +122,15 @@ class SCMRelease(AuditedModel):
     to_hash = models.CharField(max_length=64)
     # points to the pipeline run that created the release, and contains the information on the deployment pipeline
     scm_pipeline_run = models.ForeignKey(SCMPipelineRun, on_delete=models.PROTECT)
+
+
+class ApplicationMetadata(AuditedModel):
+    key = models.CharField(max_length=50)
+    value = models.CharField(max_length=200)
```
Comment (JeroennC) [inline katka/models.py]: How did you pick these max lengths?

File: tests/integration/test_application_metadata_view.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+    def test_non_existent_key(self, client, logged_in_user, application, metadata):
+        url = f'/applications/{application.public_identifier}/metadata/wrong_key/'
+        response = client.get(url)
+        assert response.status_code == 404
+
+    def test_not_my_metadata(self, client, logged_in_user, not_my_application, not_my_metadata):
+        url = f'/applications/{not_my_application.public_identifier}/metadata/{not_my_metadata.key}/'
+        response = client.get(url)
+        assert response.status_code == 404
```
Comment (JeroennC) [inline tests/integration/test_application_metadata_view.py]: I guess this happens everywhere so far, but should this be a 404? and not a 403?

File: katka/admin.py (hunk 1 above)
Comment (rodgomes) [reply katka/admin.py]: that was a merge mistake 😮

File: katka/models.py (hunk 2 above)
Comment (rodgomes) [reply katka/models.py]: randomly based on the secrets

Comment (rodgomes) [reply katka/models.py]: As we don't know what exactly we are going to put there, any number is a guess I think

File: tests/integration/test_application_metadata_view.py (hunk 3 above)
Comment (rodgomes) [reply tests/integration/test_application_metadata_view.py]: it's a 404 because you don't know that a given data from someone else exists. If you actually return 403 you will acknowledge that a wrong data exists but it's not yours. So, this can give a lot of insights for malicious people

File: katka/models.py (hunk 2 above)
Comment (JeroennC) [reply katka/models.py]: That's true. While we are pre v1 I guess it doesn't matter too much. In my head they might also become json blobs, but we can also always swap to a proper KV database if this becomes necessary
